### DISCUSSION TITLE
Don't understand the values of the counter

### DISCUSSION THREAD

**Discussion by TristanFAURE** [r-simmer/simmer#276] [Q&A] [answered]
Hello all
thank you for providing this library it is quite helpful

I'm trying to simulate a patient flow based on the tutorial "Plotting from resource monitors"

You can find bellow the code but basically I have several (around 50 by month) patients by month, one queue, and patients stayed for an average period of 72h in one bed and I want to simulate several scenario changing the numberOfBeds or TimeInHopital vars

If i perform the simulation with 15 beds the curves seems OK and I see how beds (resources) are used :
![image](https://user-images.githubusercontent.com/4500754/160397716-fdc7fd8d-2aa4-44ef-a4f6-400cd6062461.png)

If I performe the simulation with 5 beds I see that resource usage explodes :
![image](https://user-images.githubusercontent.com/4500754/160397791-b139c5de-3aeb-4a35-8944-b9729c1d98ce.png)

I would expect to have my resource usage limited by my number of  beds (15 or 5), but maybe there is something I did not get from DES theory

Thank you in advance !

```r
library(simmer)
library(simmer.plot)

timeInHopital <- 3 * 24 # mean, hours
numberOfBeds <- 5
nbMonth <- 12
numberOfPatients <- 50 * nbMonth
nbHours = nbMonth * 31 * 24

set.seed(1234)

bank <- simmer()

customer <-
  trajectory("Customer's path") %>%
  set_attribute("start_time", function() {now(bank)}) %>%
  seize("counter") %>%
  timeout(function() {rexp(1, 1/timeInHopital)}) %>%
  release("counter")

bank <-
  simmer("bank") %>%
  add_resource(name = "counter",capacity = numberOfBeds) %>%
  add_generator("Customer", customer, function() {c(0, rexp(numberOfPatients, 1/10), -1)})

bank %>% run(until = nbHours)
#> simmer environment: bank | now: 248.706140163915 | next: 
#> { Monitor: in memory }
#> { Resource: counter | monitored: TRUE | server status: 0(1) | queue status: 0(Inf) }
#> { Source: Customer | monitored: 1 | n_generated: 20 }

plot(bank,
     what = "resources",
     metric = "usage",
     names = "counter",
     items = "system",
     steps = TRUE)
```

**Comment by Enchufa2** [comment, marked as answer]
The `add_resource` function provides an infinite queue by default. The capacity is the number of beds you defined (the number of "servers"), and, when you plot "system", you see server + queue, so you see your patients accumulate in the queue. If you want your patients to be dropped from the simulation, you must limit the queue, or even remove it completely (setting `queue_size=0`).

**Comment by TristanFAURE** [comment]
Got it, it is clearer ! Thank you for the quick answer
I will try to do some changes on queue_size
